src/openforms/submissions/models/submission_value_variable.py

```diff
@@ -289,3 +293,57 @@ def __str__(self):
                 name=self.form_variable.name
             )
         return _("Submission value variable {key}").format(key=self.key)
+
+    def to_python(self) -> Any:
+        """
+        Deserialize the value into the appropriate python type.
+
+        TODO: for dates/datetimes, we rely on our django settings for timezone
+        information, however - formio submission does send the user's configured
+        timezone as metadata, which we can store on the submission/submission step
+        to correctly interpret the data. For the time being, this is not needed yet
+        as we focus on NL first.
+        """
+        if self.value is None:
+            return None
+
+        # can't see any type information...
+        if not self.form_variable_id:
+            return self.value
+
+        # we expect JSON types to have been properly stored (and thus not as string!)
+        data_type = self.form_variable.data_type
+        if data_type in (
+            FormVariableDataTypes.string,
+            FormVariableDataTypes.boolean,
+            FormVariableDataTypes.object,
```

Comment:
I'm not sure yet how we'll deal with recursion if there's a stringified datetime inside of an object, but let's tackle that when we actually run into such a situation.